testing/file_test/file_test_base.h

```diff
@@ -148,12 +132,29 @@ class FileTestBase : public testing::Test {
   static auto TransformExpectation(int line_index, llvm::StringRef in)
       -> testing::Matcher<std::string>;
 
-  const std::filesystem::path* path_;
+  const std::filesystem::path path_;
+};
+
+struct FileTestFactory {
+  // The test fixture name.
+  const char* name;
+  // A factory function for tests.
+  std::function<FileTestBase*(const std::filesystem::path& path)> factory_fn;
 };
 
 // Must be implemented by the individual file_test to initialize tests.
-extern auto RegisterFileTests(
-    const llvm::SmallVector<std::filesystem::path>& paths) -> void;
+//
+// We can't use INSTANTIATE_TEST_CASE_P because of ordering issues between
+// container initialization and test instantiation by InitGoogleTest, but this
+// also allows us more flexibility in execution.
```

Comment:
Maybe add a comment suggesting the use of the macro below?

```suggestion
//
// We can't use INSTANTIATE_TEST_CASE_P because of ordering issues between
// container initialization and test instantiation by InitGoogleTest, but this
// also allows us more flexibility in execution.
//
// We provide a `CARBON_FILE_TEST_FACTOR` macro below that provides a common,
// convenient way to implement this function.
```